test/integration/node-specific/convert_socket_errors.test.ts

```diff
@@ -77,4 +77,65 @@ describe('Socket Errors', () => {
       expect(error, error.stack).to.be.instanceOf(MongoNetworkError);
     });
   });
+
+  describe('when an error is thrown writing data to a socket', () => {
+    let client: MongoClient;
+    let collection: Collection<Document>;
+    let errorCount = 0;
+
+    beforeEach(async function () {
+      client = this.configuration.newClient({ monitorCommands: true });
+      await client.connect();
+      const db = client.db('closeConn');
+      collection = db.collection('closeConn');
+      const docs = Array.from({ length: 128 }).map((_, index) => ({ foo: index, bar: 1 }));
+      await collection.deleteMany({});
+      await collection.insertMany(docs);
+
+      for (const [, server] of client.topology.s.servers) {
+        //@ts-expect-error: private property
+        for (const connection of server.pool.connections) {
+          //@ts-expect-error: private property
+          const socket = connection.socket;
+          const stub = sinon.stub(socket, 'write').callsFake(function () {
+            errorCount++;
+            stub.restore();
+            throw new Error('This socket has been ended by the other party');
+          });
+        }
+      }
+    });
+
+    afterEach(async function () {
+      sinon.restore();
+      await client.close();
+    });
+
+    it('retries and succeeds', async () => {
+      const initialErrorCount = errorCount;
+      const commandSucceededEvents: string[] = [];
+      const commandFailedEvents: string[] = [];
+      const commandStartedEvents: string[] = [];
+
+      client.on('commandStarted', event => {
+        if (event.commandName === 'find') commandStartedEvents.push(event.commandName);
+      });
+      client.on('commandSucceeded', event => {
+        if (event.commandName === 'find') commandSucceededEvents.push(event.commandName);
+      });
+      client.on('commandFailed', event => {
+        if (event.commandName === 'find') commandFailedEvents.push(event.commandName);
+      });
```

Comment:
```suggestion
      client.on('commandStarted', filterForCommands('find', commandStartedEvents))
      ...
```

If you want - we do have a helper that makes this a bit cleaner

Reply:
Neat! Updating.